edu-services/gradebook-service/api/src/java/org/sakaiproject/service/gradebook/shared/GradebookFrameworkService.java

```diff
@@ -79,4 +80,31 @@
 	 */
 	public void setDefaultGradingScale(String uid);
 
+	/**
+	 *
+	 *	returns all the Available Grading Scales in the system.
+	 */
+	public List getAvailableGradingScales();
+
+	/**
+	 * Adds a new grade scale to an existing gradebook.
+	 *
+	 * @param scaleUuid
+	 *   The Uuid of the scale we want to be added to the gradebook
+	 * @param gradebook
+	 *   The gradebook with GradeMappings where we will add the grading scale.
+	 *
+	 */
+	public void saveGradeMappingToGradebook(String scaleUuid, Object gradebook);
```

Comment:
Most of the GB calls pass in the Gradebook uid, not the entire object. Could this be made to be the same? The code would then look it up when it is run. External tools won't have the Gradebook, just the uid reference to it.